src/components/buttons/button-embed-video-edit.jsx

```diff
@@ -0,0 +1,241 @@
+import ButtonIcon from './button-icon.jsx';
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+
+const KEY_ENTER = 13;
+const KEY_ESC = 27;
+
+/**
+ * The ButtonEmbedVideoEdit class provides functionality for changing text color in a document.
+ *
+ * @uses ButtonIcon
+ *
+ * @class ButtonEmbedVideoEdit
+ */
+class ButtonEmbedVideoEdit extends React.Component {
+
+    static key = 'embedVideoEdit';
+
+    static propTypes = {
+        /**
+         * The editor instance where the component is being used.
+         *
+         * @instance
+         * @memberof ButtonEmbedVideoEdit
+         * @property {Object} editor
+         */
+        editor: PropTypes.object.isRequired
+    };
+
+    constructor(props, context) {
+        super(props, context);
+
+        this.state = this.getInitialState();
+    }
+
+    /**
+     * Lifecycle. Invoked once, only on the client, immediately after the initial rendering occurs.
+     *
+     * Focuses on the link input to immediately allow editing. This should only happen if the component
+     * is rendered in exclusive mode to prevent aggressive focus stealing.
+     *
+     * @instance
+     * @memberof ButtonEmbedVideoEdit
+     * @method componentDidMount
+     */
+    componentDidMount() {
+        if (this.props.renderExclusive || this.props.manualSelection) {
+            // We need to wait for the next rendering cycle before focusing to avoid undesired
+            // scrolls on the page
+            if (window.requestAnimationFrame) {
+                window.requestAnimationFrame(this._focusVideoUrlInput.bind(this));
+            } else {
+                setTimeout(this._focusVideoUrlInput.bind(this), 0);
+            }
```

Comment:
You're totally right.

I'd even go further,  but I'll need @jbalsas to confirm ;). 

I'm not 100% sure what browser support we're giving for `alloy-editor` but if it's IE11+ (that's what used for saucelabs tests), we could directly use `requestAnimationFrame` since it's available there.